psl-core/src/main/java/org/linqs/psl/application/util/GroundRules.java

```diff
@@ -193,6 +194,123 @@ public static double getExpectedWeightedCompatibility(WeightedGroundRule groundR
 		return inc * groundRule.getWeight();
 	}
 
+	/** Compute the rounding probability function for a given truth value */
+	public static double roundingProbability(Double truthValue) {
+		return 0.25 + 0.5 * truthValue;
+	}
+
+	/**
+	 * Computes the expected weighted compatibility (1 - incompatibility)
+	 * of a WeightedGroundLogicalRule.
+	 * from independently rounding each {@link RandomVariableAtom} to 1.0 or 0.0
+	 * with probability equal to its current truth value.
+	 *
+	 * WARNING: the result of this function is incorrect if the RandomVariableAtoms
```

Comment:
Can we figure out a way around this?
I think this is too high a burden for the end user.
What is our use case? Do we want to round only one, or all the RVA?
If the later, then we can have a method that wraps this and resolves unweighted ground rules.

Reply:
Yes, we use this to round all RVA.  What does it mean to resolving unweighted ground rules?  My understanding is that the equivalence to MAXSAT does not hold in the case of unweighted rules.  This method (and others in the class) which are based on that equivalence have this restriction so that they are not used incorrectly.